slides/src/content/services.md

```diff
@@ -114,4 +114,134 @@ import { HttpModule } from '@angular/http';
 export class AppModule {}
 ```
 
-[View Example](https://plnkr.co/edit/E9xJ9nUGxOQ66AbOmGXL?p=preview)
\ No newline at end of file
+[View Example](https://plnkr.co/edit/E9xJ9nUGxOQ66AbOmGXL?p=preview)
+
+---
+
+## Handling `http` rejections
+
+To catch rejections we can use the `subscribe` operator's `onError` callback
+
+```ts
+  this.http.post(`${ BASE_URL }/auth/login`, payload)
+    .map(response => response.json())
+    .subscribe(
+      (authData) => this.storeToken(authData.id_token), // onNext function
+      (err) => console.error(err),                      // onError function
+      () => console.log('Authentication Complete')      // onComplete function
+    );
+```
+
+---
+
+## Using the `.catch` operator
+
+- Using `.catch` we can inspect an error and decide how to handle it
+- By not calling `subscribe` our code is more declarative
+- Can take advantage of the `async` pipe in your template
+
+```ts
+  search(term: string) {
+    return this.http.get(`https://api.spotify.com/v1/dsds?q=${term}&type=artist`)
+      .map(response => response.json())
+      .catch(e => {
+        if (e.status >==  500) return cachedVersion();
+
+        return Observable.throw(
+          new Error(`${ e.status } ${ e.statusText }`)
+        );
+      }
+  });
+```
+
+---
+
+## Cancel a Request
+
+- One of the greatest benefits to using observables over promises is the ability to cancel `http` requests
+
+```ts
+  search() {
+    const request = this.searchService.search(this.searchField.value)
+      .subscribe(
+        result => { this.result = result.artists.items; },
+        err => { this.errorMessage = err.message; },
+        () => { console.log('Completed'); }
+      );
+
+    setTimeout(() => {request.unsubscribe()}, 0); // cancel request after 0 milliseconds
+  }
+```
+[View Example](http://plnkr.co/edit/XQL8v9?p=preview)
```

Comment:
In the plunker, could you move the css in this snippet to the `styles` property?

```html
<p style="background: tomato; color: white; padding: 1rem;"
  *ngIf="errorMessage">
    {{ errorMessage }}
</p>
```